### PR TITLE
ART Summary Form With Tabs

### PR DESCRIPTION
**Pull request by dbaluku**
https://app.asana.com/0/0/1136497574562695/f

### CODE REVIEW

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 1]
```diff
@@ -0,0 +1,2169 @@
+<htmlform formName="HMIS 122a HIV Care ART Card - Summary Form"
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku new name HMIS HIV 003: HIV CARE/ART CARD

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 2]
```diff
@@ -0,0 +1,2169 @@
...
+          formDescription="Summary entry form for patients enrolled for ART"
+          formEncounterType="8d5b27bc-c2cc-11de-8d13-0010c6dffd0f"
+          formUuid="52653a60-8300-4c13-be4d-4b746da06fee"
+          formVersion="1.2"
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku change version to 3.0

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 3]
```diff
@@ -0,0 +1,1732 @@
...
+
+            if (submitValStage.indexOf(false) != -1) {
+                submitVal = false;
+                jq().toastmessage('showErrorToast', "  There are errors during the validation of the data on the form. Please correct them and submit again");
+            }
+
+            return submitVal;
+        });
+
+        function enable_disable_ei_final_status(selected_option) {
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku Remove this function as the section has changed

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 4]
```diff
@@ -0,0 +1,1732 @@
...
+                        /* remove the fade on the fields*/
+                        group.find("input").fadeTo(250, 1);
+                    }
+                }
+
+            });
+
+        }
+
+        function enable_disable_ei_pcr_result(selected_option) {
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku Remove this function as this section has changed

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 5]
```diff
@@ -0,0 +1,1732 @@
...
+                        if (checked == "checked") {
+                            enableFn();
+                        } else {
+                            disableFn();
+                        }
+                    });
+                });
+
+
+                jq(".checkboxGroup").each(function () {
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku this function should be generalized in the poc.js file

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 6]
```diff
@@ -0,0 +1,1732 @@
...
+                    });
+
+                    group.find("#checkboxCheckbox").click(
+                        function () {
+                            uncheckRadioAndAll();
+                        }
+                    );
+                });
+
+                var $allTables,
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku What happens when you have multiple tables, how does this function handle that?

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 7]
```diff
@@ -0,0 +1,1732 @@
...
+        }
+    </script>
+
+    <script type="text/javascript">
+        function val() {
+            document.getElementById("fgh").disabled = "true";
+        }
+    </script>
+
+    <style type="text/css">
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku These classes should be generalized in the project CSS file

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 8]
```diff
@@ -0,0 +1,1732 @@
...
+                </li>
+
+                <li>
+                    <a href="#drug-allergies-and-art-care">
+                        Drug Allergies &amp; ART Care
+                    </a>
+                </li>
+            </ul>
+
+            <div id="care-entry-section" style="background: #F7F7F7;">
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku Remove inline CSS background color - add a class art-care-card (with the background color)

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 9]
```diff
@@ -0,0 +1,1732 @@
...
+            width:200px;
+        }
+    </style>
+
+    <!-- Beging of HTML Form -->
+    <div class="ai-page-frame">
+        <div class="headers" style="text-align: center;">
+            <h3>HIV Care/ART CARD - Summary Page</h3>
+        </div>
+        <div id="tabs">
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku Are these bootstrap based tabs?

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 10]
```diff
@@ -0,0 +1,1732 @@
...
+                        <div class="form-group col-md-2" required="true">
+                            <label><span class="required">*</span>Pt Clinic #:</label>
+                        </div>
+                        <div class="form-group col-md-2">
+                        <patient field="identifier" identifierTypeId="e1731641-30ab-102d-86b0-7a5022ba4115" required="false"/>
+                        </div>
+                    </div>
+
+                    <div class="row">
+                        <div class="form-group col-md-2">
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku as discussed with @solemabrothers do not use col-md-* classes just use col-* classes for compatibility across all device classes 

This should also go into the Technical Guide - Form Development - Bootstrap Styling Guidelines

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 11]
```diff
@@ -0,0 +1,1732 @@
...
+                            <div class="form-group col-2">
+                                <span class="ARTCareEntryPointEnableDisableChild1">
+                                    <obs conceptId="99115" labelText=""/>
+                                </span>
+                            </div>
+                        </obsgroup>
+                    </div>
+                </section>
+                <section>
+                <table>
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku This table has no rows or cells - is it misplaced

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 12]
```diff
@@ -0,0 +1,1732 @@
...
+                </table>
+                </section>
+            </div>
+
+            <section id="art-care-cohort-section">
+                <section>
+                    <h3>ART Care Start</h3>
+                    <section>
+                        <table border="0" id="art-care">
+                            <tbody style="display:block;">
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku use Bootstrap styles and approaches - see @solemabrothers and @semujju use of cards to have a consistent approach

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 13]
```diff
@@ -0,0 +1,1732 @@
...
+                            <tr>
+                                <obsgroup groupingConceptId="99162">
+                                    <td rowspan="3" style="width:10%;" class="ARTStartEnableDisable1">
+                                        <obs id="99161" conceptId="99161" labelText="At Start ART:"/>
+                                    </td>
+                                    <td rowspan="3" class="ARTStartEnableDisableChild1" style="width:10%;">
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku Global remove all inline styles - replace with Bootstrap styles

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 14]
```diff
@@ -0,0 +1,1732 @@
...
+                                    </excludeIf>
+                                </td>
+                                <td></td>
+                            </tr>
+                            </tbody>
+                        </table>
+                    </section>
+                    <section>
+                        <h4>ART Transfer In</h4>
+                        <table class="multi">
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku Use Boostrap styles

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 15]
```diff
@@ -0,0 +1,1732 @@
...
+
+                                            </td>
+                                            <td></td>
+                                            <td align="center">
+                                                <a href="#" class="add">Add</a> | <a href="#" class="remove">Remove</a>
+                                            </td>
+                                        </tr>
+                                    </obsgroup>
+                                </template>
+                                <render/>
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku Why the duplicated renders?

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml [hunk 16]
```diff
@@ -0,0 +1,1732 @@
...
+                                                <obs conceptId="90247" answerConceptIds="90046,90047,99889"
+                                                     answerLabels="6.Clinical, 7.Immuno, 8. Virological"/>
+                                            </td>
+                                            <td align="center">
+                                                <a href="#" class="add">Add</a> | <a href="#" class="remove">Remove</a>
+                                            </td>
+                                        </tr>
+                                    </obsgroup>
+                                </template>
+                                <render/>
```
Comment (ssmusoke) [inline omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: @dbaluku There is a looping mechanism for rendering the template multiple times see TB card

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 12 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 2 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 1 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 4 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 3 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 5 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 9 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Added bootstrap tabs

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 10 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Done

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 15 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: That is the implementation used even in  the TB form

File: omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml (hunk 16 above)
Comment (dbaluku) [reply omod/src/main/webapp/resources/htmlforms/122a-HIVCare_ARTCard-SummaryPage.xml]: Its the one I used

**Comment by ssmusoke** [comment]
@dbaluku First of all I have to say excellent work, my comments are styling and flow 

1. Care Entry and ART Start Registration 
  * Date of Registration at this site is the encounter date not Enrollment in HIV Care Transfer in From
  * Special Categories - consider adding 4 rows of 3 fields each (since Others Specify already forces a 4th row) 
2. Left padding on Care Entry and family is not consistent, I think we should use the one in the CARE ENTRY & ART START REGISTRATION section, so that they do not touch 

3. Family Tracking Section
  * Age and Age Unit - needs to be validated 
  * Add text is wrapped, I suggest you use a green + icon with a circle used in Ref App 
4. ART Care Cohort -change to ART Care (there is actually no cohort information) 
5. Switch/Substitution to 2nd and 3rd line must include a checkbox to confirm whether its a substitution or switch 
6. Add a calendar icon to all dates 
7. Treatment Interruptions - rename to "Treatment Interruptions & Outcomes"
8. Drug Allergies tab rename to "Drug Allergies, Conditions and Screening"

**Pull request merged**